### DISCUSSION TITLE
A Question about "nest writes" and the `transaction ` api.

### DISCUSSION THREAD

**Discussion by notzheng** [prisma/prisma#6144] [Q&A] [answered]
I'll create a fictional scenario that may not make sense, but it's a simplistic version of a module in my project.

The database model like this.

```prisma
model User {
    id                      int
    balance             int 
    chargeRecords ChargeRecord[]
}

model ChargeRecord {
    id                        int
    amount               int
    beforeBalance    int 
    afterBalance       int 
    user                    User
    userId                 Int
}
```

When a user recharges, I want to create a record, which will record the balance before and after recharge.

I just found I can't do this with transactional guarantees.

One, I can't use the nest writes.

```javascript
    const user = await prisma.user.update({
              where: {
                id: 1,
              },
              data:{
                balance: {
                    increment: 100
                },
               chargeRecords:{
                create:[
                        {
                            amount: 100,
                            afterBalance: "I can't get the update result...",
                            beforeBalance: "if I can get the update result, I can use `user.afterBalance - amount`",
                        }
                ]
                }
              },
             
})

```

And I can't do this also by `transaction` api.

```javascript
    const updateUserTask = prisma.user.update({
              where: {
                id: 1,
              },
              data:{
                balance: {
                    increment: 100
                },
     });
     const createPayRecordTask = prisma.chargeRecord.create({
        data: {
                afterBalance: "I can't get the update result also!!!"
            }
     });
     
 await prisma.$transaction([
   updateUserTask,
    createPayRecordTask
])
            
```

so, Is there any other way I can do it?

**Comment by ryands17** [comment, marked as answer]
I think you would need to use a [raw query](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) in this case as I cannot see a way to do this in a single transaction.

**Reply by notzheng** [reply]
Thank you for your reply!
Have you considered making the transaction api support return the result of an operation inside ? Just like the [sequelize](https://sequelize.org/master/manual/transactions.html#managed-transactions).

**Reply by ryands17** [reply]
We have a blog post on how [we will support transactions](https://www.prisma.io/blog/how-prisma-supports-transactions-x45s1d5l0ww1) and your case will be covered using [Optimistic Concurrency Control](https://www.prisma.io/docs/guides/prisma-guides/prisma-client-transactions-guide#optimistic-concurrency-control) once we support [updating relations](https://github.com/prisma/prisma/issues/3143) in an `updateMany` call.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
